docs/docs/spec.md

```diff
@@ -1,37 +1,31 @@
 ---
-title: Specification
+title: Nullness Specification (draft)
 ---
 
-# Specification (draft)
+# Nullness Specification (draft)
 
-This document is our draft specification for the semantics of a set of nullness
-annotations.
+This document is a draft specification for the precise semantics of our set of
+annotations for nullness analysis.
 
---------------------------------------------------------------------------------
-
-:::note Temporary advice to readers (non-normative)
+:::note Advice to readers (non-normative)
 
-For someone new to our nullness annotations, this document does not make a good
-introduction. This document is targeted more at tool authors or advanced users.
-New users will prefer to start with our [User Guide]. We are working on further
-user documentation, including Javadoc.
+The primary audience for this document is the authors of analysis tools. Some
```

Comment:
Might read better flip flopped:

"Analysis tool owners are the primary audience for this document."

Reply:
For some reason that reads weirder to me. So uh... sword fight to the death? :-)

Reply:
An unnecessarily long argument for the suggestion:

Argument from authority? See bullet 24. in section 1. https://jmlr.csail.mit.edu/reviewing-papers/knuth_mathematical_writing.pdf Technically, the advice is scoped to opening paragraphs and you could argue this is a slightly different case, but I feel it generalizes. The form `Predicate ... Subject` is more awkward in English than `Subject .... Predicate`. Subject-Verb-Object order is the default, after all. I think this reads better when you take the more specific class as the subject of the sentence when it comes to conveying intention. In this case I think the predication is the relation of some class of things (analysis tool owners) to the document, and not the relation of an abstract concept (the document audience) to a concrete class of things (analysis tool owners). I think we should try to make the concrete aspects of meaning the "centrifugal force" of the sentence; when you place the concrete portion in the S position of SVO, it adds emphasis and will tend to parse better. I take `analysis tool authors` as the subject since it's the answer to the question form: `who is the primary audience of this document?`. Existential statements like this one are a bit fuzzier though, so I do think it's more debatable than other clear-cut cases of using passive voice when active voice ought to be preferred.

Also, I didn't mean to rewrite your wording, I commented too quickly:

`Analysis tool authors are the primary audience for this document.`

You could also use `of` instead of `for`.

At any rate, I don't think it will matter much in the grand scheme of things, so I'm fine if you stick with your original wording.

Reply:
You're giving me the "I'm fine" out, so I'm gonna take it. :-)

This is really not an "opening paragraph", and if we need to set it off in a colored box or something to make that clear we can do it.

Knuth's examples both sound better the way he rewrote them. That's argument enough for me. This one, to me, really doesn't.

Part of it is my informal sorta "ADHD writing principles", which is that readers' minds are eager to tune out your words as much as possible. So I try to order sentences and paragraphs in the order "who cares about this / when they care about it --> what they care about". Here, if I start reading "Analysis tool authors", it's a subtle clue that I don't have to care about the rest because that's not me.

Reply:
I agree with Kevin's information ordering (and his dispute-resolution mechanism :))

The much too long article I've cited for this before is [this one](https://www.e-education.psu.edu/styleforstudents/c10_p6.html). It advocates for putting the "old" information at the end of the sentence and the "new" at the end. (Here's [a shorter page that makes a similar argument](https://www.student.unsw.edu.au/using-old-new-information-order-sentence.), which I just pulled from a search now without reading in any detail.)

Here, I see the old information as "We're talking about this doc" and the new information as "The doc we're talking about? It's useful for tool authors."

I think this fits well with Kevin's argument that, if the paragraph starts off by mentioning "tool authors," other people will tune out because the paragraph doesn't seem to be "about" the audience; it appears to be "about" tool authors.

(I was a huge fan of the article I linked above when I first found it, so I have surprisingly strong opinions on this :))

Reply:
Thanks! I think the information order argument is a good one, people certainly skim a lot, so it makes sense. The grammatical structure still sounds odd and robotic to me for some reason, but I think the arguments you both give are more compelling in this case.